### PR TITLE
language: support multi language

### PR DESCRIPTION
**Pull request by qile222**
- adjust the document structure, support multi language
- add English version

### CODE REVIEW

File: LANGS.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Languages
+
+* [English](en/)
+* [中文](zh-hans/)
```
Comment (yorkie) [inline LANGS.md]: Corrections: 1) 中文 > 简体中文 2) zh-hans > zh-CN

File: README.md [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
+
+- improve the description.
+- translations.
+- inconsistence between implementation and this book.
+- typos and mistakes.
+- errors on sample.
+
+### License
+
+[Apache 2.0][]
```
Comment (yorkie) [inline README.md:41]: Please remove this line.

File: book.json [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+          "width": "160",
+          "size": "small"
+        }
+      ]
+    },
+    "disqus": {
+      "shortName": "YODAOS"
+    }
+  }
+}
```
Comment (yorkie) [inline book.json:40]: EOF

File: en/best-practice/01-testing.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+Functional Testing We offer two methods, one is to verify the voice interaction through the VUI, and the other is to verify by the mock tool.
+
+> ❕ Note: Functional test requires network status to be normal
+
+### VUI
+
+First, install the app to the device through the tool;
+
+```bash
+# tools/runtime-install
```
Comment (yorkie) [inline en/best-practice/01-testing.md]: Replace the # > $

File: en/best-practice/01-testing.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+Second, restart vui;
+
+```bash
+# tools/runtime-op --vuid restart
```
Comment (yorkie) [inline en/best-practice/01-testing.md]: Ditto

File: en/best-practice/01-testing.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+Finally, start the interaction and perform a functional test.
+
+```
+For example: Ruo Qi, I want to listen to children's songs.
```
Comment (yorkie) [inline en/best-practice/01-testing.md]: I think "Ruo Qi" should still be in Chinese, or in pinyin with tones.

File: en/best-practice/01-testing.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      @param {object} [options]
+      runtime.onVoiceCommand('asr', {intent: 'play_song'}, {appId: 'appdemo'}, {})
+      // emit app url event
+      @param {string} url
+      @param {object} [options]
+      @param {'cut' | 'scene'} [options.form='cut']
+      @param {boolean} [options.preemptive=true]
+      @param {string} [options.carrierId]
+      @returns {Promise<boolean>}
+      runtime.openUrl('url', {form: 'cut'})
```
Comment (yorkie) [inline en/best-practice/01-testing.md]: The above lines crash the program :(

File: en/best-practice/01-testing.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+## Stability Testing
+
+1. Perform a stability test on your application by executing monkey.
+
+For example, the test object is a music application, which can be tested by calling the mock tool.
+
+```bash
+# tools/mock --asr 'I want to listen to Jay Chou's songs'
+# tools/mock --asr 'I want to listen to Daoxiang'
+# tools/mock --asr 'change one'
```
Comment (yorkie) [inline en/best-practice/01-testing.md]: Ditto, use $ instead of #.

File: en/best-practice/03-performance-stability.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+
+First let's take a look at the runtime of YODAOS: YODAOS is based on [ShadowNode][] which uses an event-driven, non-blocking I/O model; at the beginning of the design, the [ShadowNode][] interface is compatible with [Node.js][], So in most scenarios, developers can use [ShadowNode][] like [Node.js][] to understand that these will help developers to develop applications on YODAOS more quickly.
+
+When developing applications, YODAOS needs to pay attention to the performance and stability of the application, including but not limited to the following:
+
+- Quick Start
+- Quick response to voice interaction
+- No exceptions and crashes
+
+## Start Up
```
Comment (yorkie) [inline en/best-practice/03-performance-stability.md]: Should be "bootstrap"

File: en/best-practice/03-performance-stability.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+The main thread of the application is mainly responsible for receiving and processing system events (NLP, buttons, etc.), so the main thread is also generally called the UI thread. Due to the particularity of the main thread itself, if the application includes blocking operations such as I/O, the application will not be able to respond to system events in time, which will result in poor user experience; while [ShadowNode][] is not thread-safe programming. Model, so don't manipulate [ShadowNode][] and its related APIs in other threads.
+
+As mentioned earlier, [ShadowNode][] uses an event-driven, non-blocking I/O model. This model is implemented via [libtuv][]. If the application logic contains I/O or other blocking tasks, the developer can Put the task into the thread pool of [libtuv][]. After the execution, it will call back in the main thread. You don't need to handle the logic such as thread synchronization yourself. For examples, please refer to [Official Implementation](https://github.com/ Libuv/libuv/blob/e4087dedf837f415056a45a838f639a3d9dc3ced/docs/code/queue-work/main.c).
```
Comment (yorkie) [inline en/best-practice/03-performance-stability.md]: "the developer can Put the task" -> The Put should be in lower case :)

Comment (yorkie) [reply en/best-practice/03-performance-stability.md]: And the link is wrong.

File: en/best-practice/03-performance-stability.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+Applications that are in ANR will not be able to receive and process user input, which is very bad for the user experience, so applications should avoid this situation. Here are some common scenarios that can lead to ANR:
+
+- Synchronize I/O on the main thread
+- Do a lot of intensive calculations on the main thread for a long time
+- The main thread synchronizes waiting for the processing results of other threads, while other threads do not process them in time, such as `thread.join()` or condition variables.
+- The main thread and other threads form a deadlock
+
+The common solutions are listed below:
+
+- For a large number of intensive calculations, you can use the tools provided by the system such as `simpleperf` to view the application function call for a period of time (requires unstriped libraries); if the CPU is mostly consumed in the virtual machine, you can pass [ShadowNode][] Provided [CPU Profiler](https://github.com/Rokid/ShadowNode/blob/bc244fe51236ddc70a3fae85a888594d99fd8e7f/docs/devs/Optimization-Tips.md#cpu-profiler) to generate [flame map](http://www.brendangregg.com/flamegraphs.html) to view script function calls
```
Comment (yorkie) [inline en/best-practice/03-performance-stability.md]: 1. flame map -> flamegraph

File: en/best-practice/03-performance-stability.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+  setInterval(() => {
+    var timestamp = Date.now()
+    obj[timestamp] = true
+  }, 1000)
+  ```
+
+- No release after creating or getting an object using [N-API][]
+
+  ```c
+  Napi_value functionExportToJS(napi_env env, napi_callback_info info) {
```
Comment (yorkie) [inline en/best-practice/03-performance-stability.md]: The `Napi_value` should be `napi_value`

File: en/best-practice/03-performance-stability.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+    napi_value value = argv[0];
+    napi_ref ref = NULL;
+    napi_create_reference(env, value, 1, ref);
+    // balabalabala...
+    // napi_delete_reference(env, ref);
+    return NULL;
+  }
+  ```
+
+- Multi-threaded callback is not enabled [Handle Scope](https://nodejs.org/docs/latest/api/n-api.html#n_api_napi_open_handle_scope)
```
Comment (yorkie) [inline en/best-practice/03-performance-stability.md]: Not enabled -> disabled.

File: en/guidance/01-build-your-first-app.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
+## YODAOS Developer Tools
+
+[YODAOS Developer Tools](https://github.com/yodaos-project/yoda-platform-tools) provides a range of convenient tools for manipulating YODAOS devices, such as sending text commands, sending NLP intents, launching, Install applications, etc.
```
Comment (yorkie) [inline en/guidance/01-build-your-first-app.md]: Install > installing

**Review by yorkie — CHANGES_REQUESTED** [review]
Too much errors, please fix them firstly

**Comment by yorkie** [comment]
And we need a directory: `locals` and put different versions inside there.

File: en/yodaos-source/customization/03-light.md [hunk 15]
```diff
@@ -112,21 +112,25 @@ The number of led lights can be controlled by `BOARD_LED_NUMS`
...
-           .tag = HARDWARE_MODULE_TAG,
-           .module_api_version = LEDS_API_VERSION,
-           .hal_api_version = HARDWARE_HAL_API_VERSION,
-           .id = LED_ARRAY_HW_ID,
-           .name = "ROKID PWM LEDS HAL",
-           .methods = &led_module_methods,
-       };
+
+```c
+ Static struct hw_module_methods_t led_module_methods = {
```
Comment (yorkie) [inline en/yodaos-source/customization/03-light.md]: Static -> static

File: en/yodaos-source/customization/03-light.md [hunk 16]
```diff
@@ -112,21 +112,25 @@ The number of led lights can be controlled by `BOARD_LED_NUMS`
...
+     .open = led_dev_open,
+ };
+
+ Struct hw_module_t HAL_MODULE_INFO_SYM = {
```
Comment (yorkie) [inline en/yodaos-source/customization/03-light.md]: Ditto

Comment (guoguo0416) [reply en/yodaos-source/customization/03-light.md]:
```suggestion
 struct hw_module_t HAL_MODULE_INFO_SYM = {
```

File: en/guidance/01-build-your-first-app.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+module.exports = function main (activity) {
+
+}
+```
+
+The application interacts with the system through the `activity` object provided by the system framework, such as receiving the system, application events, and calling the system API.
+
+### Interact with YODAOS
+
+The `activity` provided by the system framework is an object that conforms to Node.js [EventEmitter API](https://nodejs.org/docs/latest/api/events.html#events_events), on which we can listen to any application. Life cycle events:
```
Comment (guoguo0416) [inline en/guidance/01-build-your-first-app.md]: 'activity' that main function param should provide all supported api list

File: en/best-practice/03-performance-stability.md (hunk 9 above)
Comment (guoguo0416) [reply en/best-practice/03-performance-stability.md]: this file 'Official Implementation' link do not work, and there are more one place using it 
plz fix it

File: en/yodaos-source/system/compile-run.md [hunk 18]
```diff
@@ -0,0 +1,345 @@
...
+
+```sh
+$ ./build.sh -p leo_k18_universal_node -n openwrt-leo-k18-universal -f leo-k18-universal -j32 -r
+```
+
+Method 3 is to call method 2 and compile and generate the Recovery partition and the OTA firmware, which is usually used for production image publishing.
+
+#### Firmware Address
+
+* Refer to Chapter 6 of the Rokid Kamino18 Universal Install Guide V2.5.pdf for a brief description.
```
Comment (guoguo0416) [inline en/yodaos-source/system/compile-run.md]: using 'Refer to ...' should add link to the reference page
plz fix it

File: zh-hans/yodaos-source/system/compile-run.md [hunk 19]
```diff
@@ -197,7 +197,7 @@ $ ./download.sh all
 
 ### 定制系统和应用
 
-YodaOS 是基于 openwrt 编译系统，所以需要了解优化后的目录结构。	
+YODAOS 是基于 openwrt 编译系统，所以需要了解优化后的目录结构。	
 
 #### OpenWrt 目录结构
```
Comment (guoguo0416) [inline zh-hans/yodaos-source/system/compile-run.md]: this file 'please refer to [API chapter]...' link do not work, actually result is page 404
plz fix it

**Review by algebrait — APPROVED** [review]

**Pull request merged**